mypy/server/astdiff.py

```diff
@@ -57,7 +57,8 @@ def is_similar_node_shallow(n: SymbolTableNode, m: SymbolTableNode) -> bool:
     #   type_override
     if (n.kind != m.kind
             or n.mod_id != m.mod_id
-            or n.module_public != m.module_public):
+            or n.module_public != m.module_public
+            or n.module_public_even_with_underscore != m.module_public_even_with_underscore):
```

Comment:
I wasn't sure of whether this should be included here or not, since it's affected by __all__ only. I would greatly appreciate if someone who has a better idea of the use of this function could comment on this.